### ISSUE TITLE
How can I catch errors?

### ISSUE THREAD

**Issue by chrissound** [scotty-web/scotty#215]
https://github.com/chrissound/ScottyExceptionTest

```
{-# LANGUAGE OverloadedStrings #-}
module Main where

import Web.Scotty
import Control.Monad.IO.Class
import Control.Exception

main :: IO ()
main = do
  catch (
    scotty 3000 $ do
      defaultHandler (\e -> do
                        liftIO $ print "error handler reached"
                        html e)
      get "/r" $ do
        raise "this gets caught"
      get "/e" $ do
        error "this does not"
   ) (\e -> do
        print ("An exception has been caught" :: String)
        putStrLn $ show (e :: SomeException)
    )

```
`/e`
  produces HTTP output of
`    "Something went wrong"`
  produces STDout output of 
```
    this does not
  CallStack (from HasCallStack):
    error, called at /home/chris/Projects/Haskell/ScottyExceptionTest/src/Main.hs:18:9 in main:Main
```

`/r`
  produces HTTP output of
`    "this gets caught"`
  produces STDout output of 
`    "error handler reached"`

----

It seems I can't catch the error for `/e`?

**Comment by chrissound** [comment]
It seems I need to surround the `error "this does not"` with `liftAndCatchIO`. I think the documentation for `defaultHandler` is slightly misleading in this case.

**Comment by Akii** [comment]
This works until you use scottyT. Then it becomes really awkward. I feel like scotty's types are missing some "modern" type classes? MonadThrow MonadCatch and the like.

**Comment by elrikdante** [comment]
Would liftAndCatchIO from here help?

https://hackage.haskell.org/package/scotty-0.11.0/docs/Web-Scotty-Trans.html#t:ScottyError

I use it when interleaving database connections.

**Comment by Akii** [comment]
No, I'd need something like `liftAndCatch :: (ScottyError e, MonadIO m) => m a -> ActionT e m a`.
I'll investigate this now and see if I can come up with a solution either on Scotty side or on my side.

**Comment by Akii** [comment]
So apparently Scotty on master has all that stuff but on Hackage it's missing. Which makes me a sad panda.

**Comment by RyanGlScott** [comment]
@Akii, I'll try to prepare a new release this weekend.

**Comment by Akii** [comment]
Thank you very much! Let me know if I can help you, I'm on IRC as well.

**Comment by RyanGlScott** [comment]
I've uploaded [`scotty-0.11.1`](http://hackage.haskell.org/package/scotty-0.11.1) to Hackage.

**Comment by ad-si** [comment]
@Akii Did you open a merge request? I just got bitten by this 🙈

**Comment by Akii** [comment]
Probably not, got discouraged after opening my first PR, I guess.. Seeing it's still open I feel reassured. :) I've no idea what this is even about anymore lol

**Comment by ad-si** [comment]
yeah, the project seems a little unmaintained 😕

**Comment by chessai** [comment]
this is resolved, remaining issue mentioned is on #237
